qless-java/src/main/java/com/moz/qless/Queue.java

```diff
@@ -23,6 +22,91 @@
 import redis.clients.jedis.Transaction;
 
 public class Queue {
+
+  public static class JobSpec {
```

Comment:
Ok I can get behind that.

-- 
Brandon Forehand
On Feb 9, 2015 8:25 AM, "Dan Lecocq" <notifications@github.com> wrote:

> In qless-java/src/main/java/com/moz/qless/Queue.java
> <https://github.com/seomoz/qless-java/pull/21#discussion_r24340795>:
>
> > @@ -23,6 +22,91 @@
> >  import redis.clients.jedis.Transaction;
> >
> >  public class Queue {
> > +
> > +  public static class JobSpec {
>
> Right. It's early. I do have all the setters.
>
> I actually intended this to explicitly /not/ be final, because I want to
> be able to describe something like this:
>
> public class ToolJob {
>
>   private static final String KIJI = "kiji";
>   private static final String LOCAL = "local";
>   private static final String REDUCERS = "reducers";
>   private static final String EXTRA_ARGS = "extraArgs";
>   private static final String LEVEL = "level";
>
>   public static class JobSpec extends com.moz.qless.JobSpec {
>
>     public JobSpec setKiji(final String kiji) {
>       return data(KIJI, kiji);
>     }
>
>     public JobSpec setLocal(final boolean local) {
>       return data(LOCAL, local);
>     }
>
>     ...
>
>   }
>
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/seomoz/qless-java/pull/21/files#r24340795>.
>